### PR TITLE
Cell gaussian filters and MT cleanup

### PR DESCRIPTION
**Pull request by rjanvier**
CC currently point to this branch but not to the master one.

### CODE REVIEW

File: include/DgmOctree.h [hunk 1]
```diff
@@ -1215,8 +1214,39 @@ namespace CCCoreLib
...
+		//! Octree cell description helper struct
+		struct octreeCellDesc
+		{
+			DgmOctree::CellCode truncatedCode;
+			unsigned i1, i2;
+			unsigned char level;
+		};
+
+		//! Structure containing objects needed to run octree operations in parallel
+		struct MultiThreadingWrapper
```
Comment (dgirardeau) [inline include/DgmOctree.h:1226]: Why do you bring it back to the header file? It has been (voluntarily) moved to the cpp file a while ago 😅

File: src/DgmOctree.cpp [hunk 2]
```diff
@@ -3052,16 +3023,16 @@ unsigned DgmOctree::executeFunctionForAllCellsAtLevel(	unsigned char level,
...
-			// retrieve the maximum number of threads
+			//retrieve the maximum number of threads
 			maxThreadCount = QThread::idealThreadCount();
 		}
-		if (maxThreadCount == 1)
-		{
-			// if only one thread should/cloud be used, the direct approach is more efficient
-			multiThread = false;
-		}
+		//if (maxThreadCount == 1)
```
Comment (dgirardeau) [inline src/DgmOctree.cpp:3029]: why removing this?

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: src/DgmOctree.cpp (hunk 2 above)
Comment (rjanvier) [reply src/DgmOctree.cpp:3029]: I just merge these commit, CC point to this branch https://github.com/CloudCompare/CCCoreLib/tree/52667084b18f872d737f97d677f304f962e76f53

Comment (dgirardeau) [reply src/DgmOctree.cpp:3029]: Wow, I'm lost 😅 

There have been way too many back and forth...

Comment (dgirardeau) [reply src/DgmOctree.cpp:3029]: So I guess we should merge it as is 🤔 . Let's trust me-from-the-past 😅

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**